### PR TITLE
simplewallet: add show_qr_code command /monero#6488 

### PR DESCRIPTION
**Pull request by ghost**
monero-project#6488
monero-project#6766

### CODE REVIEW

**Comment by tomato5040** [comment]
This is needed for new GUI

**Comment by BigslimVdub** [comment]
Tomato tomahto

**Comment by tomato5040** [comment]
Yes, this allows qr code in gui. I guess it is not mandatory, but is not a bad feature to include.
Screenshot from 2021-05-21 11-45-55

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -1800,6 +1802,62 @@ bool simple_wallet::version(const std::vector<std::string> &args)
...
+#else
+#define PRINT_UTF8(pre, x) std::cout << x
+#define WTEXTON()
+#define WTEXTOFF()
+#endif
+
+  WTEXTON();
+  try
+  {
+    const std::string address = "monero:" + m_wallet->get_subaddress_as_str({m_current_subaddress_account, subaddress_index});
```
Comment (stoffu) [inline src/simplewallet/simplewallet.cpp:1835]: Replace with `aeon:`

**Review by stoffu — COMMENTED** [review]
AEON GUI currently does have functionality of showing QR code.

image

**Comment by stoffu** [comment]
What do you mean by your "new GUI"?

File: external/qrcodegen/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,8 @@
 project(libqrcodegen)
 
 add_library(qrcodegen STATIC QrCode.cpp)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+set_target_properties(qrcodegen PROPERTIES POSITION_INDEPENDENT_CODE ON)
+set_target_properties(qrcodegen PROPERTIES CXX_STANDARD 11)
```
[the remark this answers is not in the document]
Comment (ghost) [reply external/qrcodegen/CMakeLists.txt:5]: Resolved.

File: src/simplewallet/simplewallet.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ghost) [reply src/simplewallet/simplewallet.cpp]: Resolved.

File: src/simplewallet/simplewallet.cpp [hunk 3]
```diff
@@ -1832,7 +1832,7 @@ bool simple_wallet::show_qr_code(const std::vector<std::string> &args)
   WTEXTON();
   try
   {
-    const std::string address = "monero:" + m_wallet->get_subaddress_as_str({m_current_subaddress_account, subaddress_index});
+    const std::string address = "aeon:" + m_wallet->get_subaddress_as_str({m_current_subaddress_account, subaddress_index});
```
Comment (stoffu) [inline src/simplewallet/simplewallet.cpp:1835]: Just a matter of taste, but it could have been done in the first cherry-pick commit. But OK.

**Comment by ghost** [comment]
I don't want this to be merged unless there is more support from other developers and contributors. Expires May 16 2022.

**Comment by BigslimVdub** [comment]
So these changes are solely for adding QR support in CLI wallet? 
If so I am for the changes pending it doesn't break any other builds (including GUI). According to the checks it did pass 13 days ago per ci builds so I would say ok to merge.

**Comment by iamsmooth** [comment]
If it is just a clean merge from Monero then why not? It is generally to our advantage to have fewer differences from upstream that aren't core to the coin design.

**Review by stoffu — APPROVED** [review]

**Pull request merged**